### PR TITLE
Add Panorama servers to general settings

### PR DESCRIPTION
**Pull request by wolffberg**
## Description

Add support for primary and secondary Panorama server(s) under general settings.

## Motivation and Context

The option to set the Panorama servers were not available and were occupied by a placeholder.

## How Has This Been Tested?

The changes has been tested using the panos Terraform module https://github.com/PaloAltoNetworks/terraform-provider-panos

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by shinmog — CHANGES_REQUESTED** [review]
Looks like the location for the primary and secondary Panorama servers was moved in PAN-OS 9.0.  So fixing this will require a new struct that is keyed to version 9.0.  The new structure for Panorama looks like this:

```xml
<sequence help-string="Centralized management system" name="panorama" optional="yes">
    <choice default="local" optional="yes">
        <sequence disabled="yes" help-string="Managed by Panorama Cloud Service" name="cloud-service"/>
        <sequence help-string="Local panorama instance" name="local-panorama">
            <element help-string="Panorama server (IP address or FQDN)" maxlen="63" name="panorama-server" optional="yes" regex="^([0-9a-zA-Z:./_-])+$" type="string"/>
            <element help-string="Panorama server 2 (IP address or FQDN)" maxlen="63" name="panorama-server-2" optional="yes" regex="^([0-9a-zA-Z:./_-])+$" type="string">
                <validate err="Primary and Secondary Panorama server IP addresses cannot be the same" expr="not(.=../panorama-server)"/>
            </element>
        </sequence>
    </choice>
</sequence>
```

Since the `cloud-service` param is `disabled="yes"` you can safely ignore that.

**Comment by wolffberg** [comment]
I'm a bit unsure where and how to do the version check. Please advice 👷‍♂️

**Comment by shinmog** [comment]
Hrm..  So, I haven't updated this namespace yet (#29)...  So, three options, hardest to easiest:

1. You can do the namespace update and the version update (you can look at stuff like `netw/routing/protocol/bgp` to see what a singleton namespace implementation looks like).
2. I'll do the namespace update, then you do the new version update
3. Ask me to do both.

Let me know what you think.

**Comment by wolffberg** [comment]
Tests are passing but I have tried reusing a few structs. Not sure if it fits into the general structure of pango or adds unnecessary complexity. What do you think?
I also need help testing as I can't make changes to our firewalls for at least a few weeks and I don't have other versions of PanOS available at this time.

File: dev/general/config.go [hunk 1]
```diff
@@ -310,50 +332,86 @@ func (o *container_v1) Normalize() Config {
...
+	Timezone              string          `xml:"timezone,omitempty"`
+	Domain                string          `xml:"domain,omitempty"`
+	UpdateServer          string          `xml:"update-server,omitempty"`
+	VerifyUpdateServer    string          `xml:"server-verification"`
+	LoginBanner           string          `xml:"login-banner,omitempty"`
+	ProxyServer           string          `xml:"secure-proxy-server,omitempty"`
+	ProxyPort             int             `xml:"secure-proxy-port,omitempty"`
+	ProxyUser             string          `xml:"secure-proxy-user,omitempty"`
+	ProxyPassword         string          `xml:"secure-proxy-password,omitempty"`
+	Panorama              *devicePanorama `xml:"panorama"`
```
Comment (shinmog) [inline dev/general/config.go:350]: I get what you're trying to do here, but I suspect that this wouldn't actually work run against a real PAN-OS with the old and new Panorama server locations...?

**Review by shinmog — APPROVED** [review]
Sorry for the delay on this :(

Ok, so thanks for all the work you put into this!  Updating the namespace to the new style was a lot of work, so I really appreciate what you've done here.  That said, the style is not consistent with the other namespaces, and I suspect that this code would not actually work in practice for both the old and new styles of where Panorama servers are located.  So I'm going to take this PR as-is because you've already done so much, but I'll be tweaking the code in this namespace to reflect my feedback comments.

In light of this PR, I definitely need to create a document that goes over contributing to pango (#65) to help people contribute in the future.

**Comment by shinmog** [comment]
Just saw that the BGP namespace has the specifier structs as `entry_vX`; lol.  So, the intent was that they be named `config_vX`.

**Pull request merged**